bin/cron/teacher_applications_to_gdrive

```diff
@@ -15,43 +16,78 @@ class WebAgent
   end
 end
 
-#
-# Uses a Google Cloud service account to write teacher application data for this year into
+# Writes teacher application data for this year into
 # a spreadsheet in Google Drive (with permissions locked down to our organization) for exploration
 # by our programs team.
-#
-
-applications = [%w(course regional_partner status scholarship_status school_type pay_fee email created_at)]
-
-Pd::Application::Teacher2021Application.find_each do |app|
-  applications << [
-    app.course,
-    app.regional_partner.try(:name),
-    app.status,
-    app.scholarship_status,
-    app.school_type,
-    app.sanitize_form_data_hash[:principal_pay_fee],
-    app.email,
-    app.created_at
+def update_sheets(drive)
+  # Update application sheet
+  applications = [%w(course regional_partner status scholarship_status school_type pay_fee email created_at)]
+  Pd::Application::Teacher2021Application.find_each do |app|
+    applications << [
+      app.course,
+      app.regional_partner.try(:name),
+      app.status,
+      app.scholarship_status,
+      app.school_type,
+      app.sanitize_form_data_hash[:principal_pay_fee],
+      app.email,
+      app.created_at
+    ]
+  end
+
+  drive.update_sheet applications, CDO.applications_2020_2021_gsheet_key, 'all_apps (auto)'
+
+  # Update metadata sheet
+  last_updated = DateTime.now.in_time_zone ActiveSupport::TimeZone.new "Pacific Time (US & Canada)"
+  metadata = [
+    ['AUTOMATION METADATA'],
+    [''],
+    ['The tabs "all_apps (auto)" and "meta (auto)" are auto-generated;'],
+    ['Any edits you make to them (besides formatting) may be lost.'],
+    [''],
+    ['Last updated:', last_updated.strftime('%Y-%m-%d %l:%M%P GMT%:::z')],
+    ['Written by:', CDO.gdrive_export_secret.client_email],
+    ['Teacher applications:', applications.length - 1],
+    [''],
+    ['Parts of this Google Sheet are auto-populated from our live application by an automated process.'],
+    ["The sheet is shared with a \"service account\" that updates it on the application's behalf."],
+    ["(Technical Details: https://github.com/code-dot-org/code-dot-org/pull/32597)"],
   ]
+
+  drive.update_sheet metadata, CDO.applications_2020_2021_gsheet_key, 'meta (auto)'
+end
+
+def notify_of_external_sharing(drive)
+  # List of external emails that we can share this document with.
+  # This list is saved in DCDO as an array. To append a new value to this list:
+  # DCDO.set(key_name, DCDO.get(key_name, []) << new_value)
+  allowed_list = DCDO.get('external_emails_with_application_data_access', []) << CDO.gdrive_export_secret.client_email
```

Comment:
will this cause any issues if 'external_emails...' isn't set yet or gets removed accidentally in the future?

Reply:
If the key doesn't exist, `DCDO.get(key_name, default_value)` will return the default value, which is an empty array in this case. So it will be fine.